scripts/scripts/vaccinations/automations/batch/israel.py

```diff
@@ -2,39 +2,110 @@
 import json
 import requests
 import pandas as pd
+from .utils.pipeline import enrich_total_vaccinations
 
 
-def main():
+def read(source: str) -> pd.DataFrame:
+    data = json.loads(requests.get(source).content)
+    df = pd.DataFrame.from_records(data)
+    return df
 
-    url = "https://datadashboardapi.health.gov.il/api/queries/vaccinated"
 
-    data = json.loads(requests.get(url).content)
+def rename_columns(df: pd.DataFrame) -> pd.DataFrame:
+    return df.rename(
+        columns={
+            "Day_Date": "date",
+            "vaccinated_cum": "people_vaccinated",
+            "vaccinated_seconde_dose_cum": "people_fully_vaccinated"
+        }
+    )
 
-    df = pd.DataFrame.from_records(data)
 
-    df = df.rename(columns={
-        "Day_Date": "date",
-        "vaccinated_cum": "people_vaccinated",
-        "vaccinated_seconde_dose_cum": "people_fully_vaccinated"
-    })
+def format_date(df: pd.DataFrame) -> pd.DataFrame:
```

Comment:
Perhaps those should be decoupled into:

```python
def format_date(df: pd.DataFrame) -> pd.DataFrame:
    return df.assign(date=df.date.str.slice(0, 10))

def filter_date(df: pd.DataFrame) -> pd.DataFrame:
    return df[df.date < str(datetime.date.today())]
```

This decouples the logic (cognitive load, reusability), and we don’t mutate inputs.